runtime/generator.go

```diff
@@ -108,18 +111,45 @@ func generatorIter(f *Frame, o *Object) (*Object, *BaseException) {
 }
 
 func generatorNext(f *Frame, o *Object) (*Object, *BaseException) {
-	return toGeneratorUnsafe(o).resume(f, None)
+	return toGeneratorUnsafe(o).resume(f, None, nil)
 }
 
 func generatorSend(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
 	if raised := checkMethodArgs(f, "send", args, GeneratorType, ObjectType); raised != nil {
 		return nil, raised
 	}
-	return toGeneratorUnsafe(args[0]).resume(f, args[1])
+	return toGeneratorUnsafe(args[0]).resume(f, args[1], nil)
+}
+
+func generatorThrow(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+	argc := len(args)
+	if argc == 1 {
```

Comment:
You can leave out the `argc == 1` and `argc > 4` checks here. If `argc < 2 || argc > 4` then the checkMethodArgs will fail because it won't match expectedTypes. The error message isn't ideal in that case, but I think the simplicity warrants it.